plugins/figura_format/figura_format.js

```diff
@@ -147,266 +101,280 @@
+				condition: () => Format === format && (Group.selected != null || Outliner.selected.length === 1),
 				click() {
 					let path = []
-					let element = Outliner.selected[0]
+					let element = Group.selected || Outliner.selected[0]
 					while (element !== "root") {
 						path.unshift(element.name)
 						element = element.parent;
 					}
 					path.unshift(Project.name || "modelName")
-					path = path.map(index => isValidLuaIdentifier(index) ? `.${index}` : `["${index}"]`)
+					path = path.map(getValidLuaIndex)
 					path.unshift('models')
-					navigator.clipboard.writeText(path.join(""))
+					let pathString = path.join("")
+					navigator.clipboard.writeText(pathString)
+					Blockbench.showQuickMessage(`Coppied "${pathString}" to the clipboard`)
 				}
 			})
-			Cube.prototype.menu.addAction('figura_copy_path', '#manage');
-			Mesh.prototype.menu.addAction('figura_copy_path', '#manage');
-			Group.prototype.menu.addAction('figura_copy_path', '#manage');
-			Toolbars.main_tools.add(
-				new Action('figura_cycle_vertex_order', {
-					name: 'Cycle Vertex Order',
-					icon: 'fa-sync',
-					category: 'edit',
-					condition: { modes: ['edit'], features: ['meshes'], method: () => Format === format && (Mesh.selected[0] && Mesh.selected[0].getSelectedFaces().length) },
-					click() {
-						Undo.initEdit({ elements: Mesh.selected });
-						Mesh.selected.forEach(mesh => {
-							for (let key in mesh.faces) {
-								let face = mesh.faces[key];
-								if (face.isSelected()) {
-									if (face.vertices.length < 3) continue;
-									[face.vertices[0], face.vertices[1], face.vertices[2], face.vertices[3]] = [face.vertices[1], face.vertices[2], face.vertices[3], face.vertices[0]];
-								}
-							}
-						})
-						Undo.finishEdit('Cycle face vertices');
-						Canvas.updateView({ elements: Mesh.selected, element_aspects: { geometry: true, uv: true, faces: true } });
+			const copyPathAnimation = new Action('figura_copy_path_animation', {
+				name: "Copy Animation Path",
+				description: "Calculates the scripting path to this Animation and copies it to the clipboard.",
+				icon: "fa-clipboard",
+				condition: () => Format === format && (Animation.selected !== null),
+				click() {
+					let path = [Project.name || "modelName", Animation.selected.name]
+					path = path.map(getValidLuaIndex)
+					path.unshift('animations')
+					let pathString = path.join("")
+					navigator.clipboard.writeText(pathString)
+					Blockbench.showQuickMessage(`Coppied "${pathString}" to the clipboard`)
+				}
+			})
+			const copyPathTextures = new Action('figura_copy_path_texture', {
+				name: "Copy Texture Path",
+				description: "Calculates the scripting path to this Texture and copies it to the clipboard. Results may vary.",
+				icon: "fa-clipboard",
+				condition: () => Format === format && (Texture.selected !== null),
+				click() {
+					let texture = PathModule.parse(Texture.selected.path)
```

Comment:
Since the PathModule variable is not defined in the web app, this action should have a condition to only work in the desktop app, or use another approach to process the path, or display an error.

Reply:
whoops, I completely forgot about the webapp. Fixed